libbeat/outputs/kafka/config_test.go

```diff
@@ -97,3 +103,32 @@ func TestConfigInvalid(t *testing.T) {
 		})
 	}
 }
+
+func TestBackoffFunc(t *testing.T) {
+	lbtesting.SeedPRNG(t)
+	tests := map[int]backoffConfig{
+		15: {Init: 1 * time.Second, Max: 60 * time.Second},
+		7:  {Init: 2 * time.Second, Max: 20 * time.Second},
+		4:  {Init: 5 * time.Second, Max: 7 * time.Second},
+	}
+
+	for numRetries, backoffCfg := range tests {
+		t.Run(fmt.Sprintf("%v_retries", numRetries), func(t *testing.T) {
+			backoffFn := makeBackoffFunc(backoffCfg)
+
+			prevBackoff := backoffCfg.Init
+			for retries := 1; retries <= numRetries; retries++ {
+				backoff := prevBackoff * 2
+
+				expectedBackoff := math.Min(float64(backoff), float64(backoffCfg.Max))
+				actualBackoff := backoffFn(retries, 50)
+
+				require.GreaterOrEqual(t, float64(actualBackoff), expectedBackoff/2)
+				require.LessOrEqual(t, float64(actualBackoff), expectedBackoff)
```

Comment:
Normally I'd prefer not to use floats, but it seems to be ok. For error reporting, maybe just use t.Fatal like this?

```
if !(expected / 2 <= actual <= expected) {
  t.Fatalf("backoff '%v' not in expected range [%v, %v] (retries: %v)", actual, expected/2, expected, retries)
}
```